pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -429,6 +431,25 @@ public AtomicLong getPendingWriteOps() {
         return pendingWriteOps;
     }
 
+    public CompletableFuture<Void> unloadSubscription(String subName) {
+        final PersistentSubscription sub = subscriptions.get(subName);
+        if (sub == null) {
+            return CompletableFuture.failedFuture(new RestException(Response.Status.NOT_FOUND,
+                    String.format("Subscription %s not found", subName)));
+        }
+        if (Compactor.COMPACTION_SUBSCRIPTION.equals(sub.getName())){
```

Comment:
Deduplication subscription?

Reply:
Since the deduplication feature just uses a cursor(will not create a subscription), it will not exist in the map `PersistentTopic.subscriptions`, so if call `unload( dedup )` will response "404 Not Found."